adapters/mqtt-vertx/src/test/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapterTest.java

```diff
@@ -257,6 +257,47 @@ public void testMapTopicSupportsShortAndLongTopicNames(final VertxTestContext ct
 
     }
 
+    /**
+     * Verifies that the adapter calls the mapper if configured. Also the updated payload and deviceId is checked.
+     *
+     * @param ctx The helper to use for running tests on vert.x.
+     */
+    @Test
+    public void testMapperShouldUpdatePayloadAndDeviceId(final VertxTestContext ctx) {
+        //TODO: change test to address mapping.
+        givenAnAdapter();
+
+        MqttPublishMessage message = newMessage(MqttQoS.AT_LEAST_ONCE, EventConstants.EVENT_ENDPOINT);
+        MqttContext context = newContext(message, null);
+        adapter.mapTopic(context).setHandler(ctx.succeeding(address -> {
+            ctx.verify(() -> assertThat(MetricsTags.EndpointType.fromString(address.getEndpoint())).isEqualTo(MetricsTags.EndpointType.EVENT));
+        }));
+
+        message = newMessage(MqttQoS.AT_LEAST_ONCE, EventConstants.EVENT_ENDPOINT_SHORT);
+        context = newContext(message, null);
+        adapter.mapTopic(context).setHandler(ctx.succeeding(address -> {
+            ctx.verify(() -> assertThat(MetricsTags.EndpointType.fromString(address.getEndpoint())).isEqualTo(MetricsTags.EndpointType.EVENT));
+        }));
+
+        message = newMessage(MqttQoS.AT_LEAST_ONCE, TelemetryConstants.TELEMETRY_ENDPOINT);
+        context = newContext(message, null);
+        adapter.mapTopic(context).setHandler(ctx.succeeding(address -> {
+            ctx.verify(() -> assertThat(MetricsTags.EndpointType.fromString(address.getEndpoint())).isEqualTo(MetricsTags.EndpointType.TELEMETRY));
+        }));
+
+        message = newMessage(MqttQoS.AT_LEAST_ONCE, TelemetryConstants.TELEMETRY_ENDPOINT_SHORT);
+        context = newContext(message, null);
+        adapter.mapTopic(context).setHandler(ctx.succeeding(address -> {
+            ctx.verify(() -> assertThat(MetricsTags.EndpointType.fromString(address.getEndpoint())).isEqualTo(MetricsTags.EndpointType.TELEMETRY));
+        }));
+
+        message = newMessage(MqttQoS.AT_LEAST_ONCE, "unknown");
+        context = newContext(message, null);
+        adapter.mapTopic(context).setHandler(ctx.failing());
+        ctx.completeNow();
+
```

Comment:
I'd rather have a single combination of qos and endpoint but have a real test instead of all combinations without any assertions.
For that, I guess the `WebClient` needs to be a field of the adapter so that we can easily use a mock in order to verify the adapter's interactions with it ...

Reply:
I agree. Test was still todo though. Will do that today and will create a field for the web client indeed.

Reply:
I added a more effective test now.